modules/sharedaddy/sharedaddy.php

```diff
@@ -188,15 +188,26 @@ function sharing_global_resources_save() {
 }
 
 function sharing_email_dialog() {
-	echo '<div class="recaptcha" id="sharing_recaptcha"></div><input type="hidden" name="recaptcha_public_key" id="recaptcha_public_key" value="'.(defined( 'RECAPTCHA_PUBLIC_KEY' ) ? esc_attr( RECAPTCHA_PUBLIC_KEY ) : '').'" />';
+	require_once plugin_dir_path( __FILE__ ) . 'recaptcha.php';
+
+	$recaptcha = new Jetpack_ReCaptcha( RECAPTCHA_PUBLIC_KEY, RECAPTCHA_PRIVATE_KEY );
+	echo $recaptcha->get_recaptcha_html(); // xss ok
 }
 
 function sharing_email_check( $true, $post, $data ) {
-	require_once plugin_dir_path( __FILE__ ).'recaptchalib.php';
+	require_once plugin_dir_path( __FILE__ ) . 'recaptcha.php';
 
-	$recaptcha_result = recaptcha_check_answer( RECAPTCHA_PRIVATE_KEY, $_SERVER["REMOTE_ADDR"], $data["recaptcha_challenge_field"], $data["recaptcha_response_field"] );
+	$recaptcha = new Jetpack_ReCaptcha( RECAPTCHA_PUBLIC_KEY, RECAPTCHA_PRIVATE_KEY );
+	$response  = '';
+	if ( isset( $_POST['g-recaptcha-response'] ) ) {
+		$response = sanitize_text_field( $_POST['g-recaptcha-response'] );
+	}
 
-	return $recaptcha_result->is_valid;
+	$result = $recaptcha->verify( $response, $_SERVER['REMOTE_ADDR'] );
+	if ( $result && ! is_wp_error( $result ) ) {
```

Comment:
Done in https://github.com/gedex/jetpack/commit/913c165c24c757a3b223bbe89597a577ac759c74.